Intern project maum 5

lifeoasis-Ian opened the pull request.

lifeoasis-Ian committed:
Intern-Project-Maum-6 [개선] Indicator 디자인 & useFocusEffect 수정

lifeoasis-Ian committed:
fixed: Create useBlockBackHandler Custom Hooks

lifeoasis-Ian committed:
fixed: Seperate Service Layer & Fix Service Logic

lifeoasis-Ian committed:
fixed: Seperate Service Layer & Fix Service Logic

lifeoasis-sera commented on src/navigation/AppNavigator.tsx:
```diff
@@ -11,31 +10,27 @@ import AuthPhoneCode from "../screens/AuthPhoneCode/AuthPhoneCode";
+
+  const navigationRef = useNavigationContainerRef<NativeStackNavigatorProps>();
+  const dispatch = useAppDispatch();
 
```
아래 useEffect 부분 언제 수정해 주실 건가요?

lifeoasis-Ian replied:
새로 브랜치 파서 작업하려고 합니다!

lifeoasis-sera commented on src/navigation/AppNavigator.tsx:
```diff
@@ -58,7 +53,7 @@ const AppNavigator: React.FC = () => {
     const isPermissionGranted =
       await permissionService.checkAndRequestPermissions();
-    const token = await AsyncStorage.getItem("token");
+    const token = await getUserDataService.getToken("token");
```
서비스 함수명이 getToken인데 인자로 token이 들어갈 필요가 있니요? 
다른 인자가 사용되는 경우 있나요?

lifeoasis-sera commented on src/screens/AuthPhone/AuthPhone.tsx:
```diff
@@ -32,56 +32,23 @@ const AuthPhone: React.FC<AuthScreenProps> = ({navigation}) => {
+      await authService.getAuthPhoneCode();
+      navigation.push("AuthPhoneCode", {phoneNumber, countryCode});
+    } catch (error: any) {
+      if (error.message === "429") {
```
에러 숫자 모아서 따로 알 수 있게 파일 만들어 주세요.

lifeoasis-sera commented on src/screens/AuthPhone/AuthPhone.tsx:
```diff
@@ -32,56 +32,23 @@ const AuthPhone: React.FC<AuthScreenProps> = ({navigation}) => {
         showAuthCodeTryOverErrorToast();
       } else {
-        navigation.push("AuthPhoneCode", {phoneNumber, countryCode});
+        console.error(error);
```
필요 없는 console 삭제해 주세요

lifeoasis-sera commented on src/screens/AuthPhone/AuthPhone.tsx:
```diff
@@ -184,7 +151,7 @@ const AuthPhone: React.FC<AuthScreenProps> = ({navigation}) => {
           keyboardType={"numeric"}
           value={phoneNumber}
-          onChangeText={onlyNum}
+          onChangeText={text => setPhoneNumber(text.replace(/[^0-9]/g, ""))}
```
onlyNum 함수 삭제하신 이유가 있을까요?

lifeoasis-Ian replied:
굳이 위에서 3줄을 차지하는 것 같아서 그냥 바로 넣었습니다!

lifeoasis-sera commented on src/screens/AuthPhoneCode/AuthPhoneCode.tsx:
```diff
@@ -61,124 +66,75 @@ const AuthPhoneCode: React.FC<AuthCodeScreenProps> = ({navigation, route}) => {
-      console.log(error);
-    }
+  const onChangeAuthCode = (code: string) => {
+    setAuthCode(code.replace(/[^0-9]/g, ""));
```
숫자만 허용되는 로직이 중복해서 나오는 것 같네요 함수로 만들어서 재사용성 높여주세요

lifeoasis-sera commented on src/screens/AuthPhoneCode/AuthPhoneCode.tsx:
```diff
@@ -61,124 +66,75 @@ const AuthPhoneCode: React.FC<AuthCodeScreenProps> = ({navigation, route}) => {
-    }
+  const onChangeAuthCode = (code: string) => {
+    setAuthCode(code.replace(/[^0-9]/g, ""));
+    setDisabled(code.length !== AUTH_CODE_CELL_COUNT);
```
code 기준으로 disabled 잡는거 그대로 있네요

lifeoasis-sera commented on src/screens/AuthPhoneCode/AuthPhoneCode.tsx:
```diff
@@ -61,124 +66,75 @@ const AuthPhoneCode: React.FC<AuthCodeScreenProps> = ({navigation, route}) => {
 
   useEffect(() => {
     if (secondsLeft > 0) {
       BackgroundTimer.runBackgroundTimer(() => {
         setSecondsLeft(secs => secs - 1);
       }, 1000);
-
       return () => BackgroundTimer.stopBackgroundTimer();
     } else {
       BackgroundTimer.stopBackgroundTimer();
     }
```
로직 한번 더 확인해 주세요

lifeoasis-sera commented on src/screens/AuthPhoneCode/AuthPhoneCode.tsx:
```diff
@@ -61,124 +66,75 @@ const AuthPhoneCode: React.FC<AuthCodeScreenProps> = ({navigation, route}) => {
+      setDisabled(true);
+      setSecondsLeft(TOTAL_TIMER_SECOND);
+    } catch (error: any) {
+      if (error.message === "429") {
```
오류 숫자 상수 처리

lifeoasis-sera commented on src/screens/AuthPhoneCode/AuthPhoneCode.tsx:
```diff
@@ -61,124 +66,75 @@ const AuthPhoneCode: React.FC<AuthCodeScreenProps> = ({navigation, route}) => {
+      if (error.message === "429") {
+        showAuthCodeTryOverErrorToast();
+      } else {
+        console.error(error);
```
불필요한 console

lifeoasis-sera commented on src/screens/AuthPhoneCode/AuthPhoneCode.tsx:
```diff
@@ -61,124 +66,75 @@ const AuthPhoneCode: React.FC<AuthCodeScreenProps> = ({navigation, route}) => {
+          navigation.push(returnPageResult);
+        }
+      } catch (error: any) {
+        if (error.message === "405") {
```
오류 상수처리

lifeoasis-sera commented on src/screens/AuthPhoneCode/AuthPhoneCode.tsx:
```diff
@@ -61,124 +66,75 @@ const AuthPhoneCode: React.FC<AuthCodeScreenProps> = ({navigation, route}) => {
+          showAuthCodeTryOverErrorToast();
         } else {
-          navigation.push("Language");
+          console.error(error);
```
필요 없는 console 빼주세요

lifeoasis-sera commented on src/screens/Language/Language.tsx:
```diff
@@ -142,14 +106,30 @@ const Language: React.FC<LanguageScreenProps> = ({navigation}) => {
+          const token = await getUserDataService.getToken("token");
+          await getLanguageUsingToken(token);
+        } catch (error) {
+          console.error(error);
```
불필요한 console 빼주세요

lifeoasis-sera commented on src/screens/Language/Language.tsx:
```diff
@@ -142,14 +106,30 @@ const Language: React.FC<LanguageScreenProps> = ({navigation}) => {
 
       setDisabled(false);
```
위에 setDisabled(true) 로직을 default로 넣기보다는 따로 빼저 하는게 좋아 보이네요

lifeoasis-sera commented on src/screens/Language/Language.tsx:
```diff
@@ -171,7 +151,7 @@ const Language: React.FC<LanguageScreenProps> = ({navigation}) => {
       await saveService.saveLanguage(changedLanguage, submitToken);
     } catch (error: any) {
-      console.log(error);
+      console.error(error);
```
불필요한 console 빼주세요

lifeoasis-sera commented on src/screens/Permission/Permission.tsx:
```diff
@@ -20,12 +17,10 @@ interface PermissionScreenProps {
-    const result = await permissionService.checkAndRequestPermissions();
-    console.log(result);
-    if (result) {
+    const resultPermissions =
+      await permissionService.checkAndRequestPermissions();
+    if (resultPermissions) {
```
함수명 어떤 권한 요청인지 알 수 있게 변경해 주세요

lifeoasis-sera commented on src/services/AuthService.ts:
```diff
@@ -1,37 +1,46 @@
     } catch (error: any) {
-      if (error.response) {
-        return error.response.status;
+      if (error.response.status === 405) {
```
status 코드 상수로 처리해 주세요

lifeoasis-Ian committed:
fixed: Rewrite Logic and Set Error Code to Enum and

- 불필요한 콘솔 제거
- getToken 함수 인자 제거
- blockStringInput 함수 생성
- AppNavigator 파일 useEffect 수정

lifeoasis-sera commented on src/screens/AuthPhoneCode/AuthPhoneCode.tsx:
```diff
@@ -37,6 +38,10 @@ interface AuthCodeScreenProps {
+export const blockStringInput = (inputString: string) => {
+  return inputString.replace(/[^0-9]/g, "");
+};
+
```
util 파일로 이동해 주세요

lifeoasis-sera commented on src/screens/Language/Language.tsx:
```diff
@@ -143,15 +133,15 @@ const Language: React.FC<LanguageScreenProps> = ({navigation}) => {
       await saveService.saveLanguage(changedLanguage, submitToken);
-    } catch (error: any) {
-      console.error(error);
+    } catch (error) {
+      throw error;
     }
```
catch해서 하는 일이 없네요

lifeoasis-sera commented on src/services/SaveService.ts:
```diff
@@ -10,16 +9,16 @@ export class SaveService {
       });
-    } catch (error: any) {
-      console.log(error);
+    } catch (error) {
+      throw error;
```
catch가 하는 일이 없어 보이네요

lifeoasis-sera commented on src/services/SaveService.ts:
```diff
@@ -10,16 +9,16 @@ export class SaveService {
       await AsyncStorage.setItem("token", token);
     } catch (error) {
-      console.log(error);
+      throw error;
```
catch가 하는 일이 없어 보이네요

lifeoasis-sera commented on src/services/index.ts:
```diff
@@ -1,7 +1,9 @@
+import {PermissionService} from "./PermissionService.ts";
 import {SaveService} from "./SaveService.ts";
+import {Platform} from "react-native";
+export let backendUrl = Platform.OS === "ios" ? "localhost" : "10.170.130.166";
```
let으로 선언한 이유가 있나요?

lifeoasis-Ian committed:
fixed/#5: Remove Try-Catch

- 불필요한 try-catch문 삭제
- 함수 utils 파일로 옮기기
